### PR TITLE
Overhauls the RPM to support open tracing protocols.

### PR DESCRIPTION
**Pull request by randy-girard**

### CODE REVIEW

**Comment by randy-girard** [comment]
Linked to https://github.com/randy-girard/app_perf/pull/18

**Comment by stereobooster** [comment]
It would be nice to use `# frozen_string_literal: true`, it will lower allocations for strings

File: lib/app_perf_rpm/instruments/sidekiq.rb [hunk 1]
```diff
@@ -3,41 +3,85 @@ class SidekiqServer
...
     def call(*args)
+      worker, msg, queue = args
+
       if ::AppPerfRpm::Tracer.tracing?
-        worker, msg, queue = args
+        operation = "Sidekiq_#{queue}##{msg["wrapped"]}"
+        span = AppPerfRpm.tracer.start_span(operation, tags: {
+          "component" => "Sidekiq",
+          "span.kind" => "client",
+          "http.url" => "/sidekiq/#{queue}/#{msg['wrapped']}",
```
Comment (stereobooster) [inline lib/app_perf_rpm/instruments/sidekiq.rb]: why `queue` and not `worker.class.name`?

**Comment by randy-girard** [comment]
I can add `# frozen_string_literal: true`. I just want to make sure everything is capable with older versions of ruby as well, which I think this should be fine with.

File: lib/app_perf_rpm/instruments/sidekiq.rb (hunk 1 above)
Comment (randy-girard) [reply lib/app_perf_rpm/instruments/sidekiq.rb]: No specific reason at this point. I have that as a `bg.job_name` tag.

**Comment by randy-girard** [comment]
Going to merge this for now, can continue to work on it. Lots of updates though.

**Pull request merged**